src/sample/Controller.java

```diff
@@ -1,4 +1,81 @@
 package sample;
 
-public class Controller {
-}
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Button;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import jodd.json.JsonSerializer;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+public class Controller implements Initializable {
+
+    ObservableList<Contact> contacts = FXCollections.observableArrayList();
+
+    @FXML
+    TextField nameSpace;
+    @FXML
+    TextField phoneSpace;
+    @FXML
+    TextField emailSpace;
+    @FXML
+    Button addButton;
+    @FXML
+    Button removeButton;
+    @FXML
+    ListView list;
+
+
+    public void addItem() {
+        //add code that makes a decision to add user only IF all three text boxes are full.....
+        //what i want is: if all text spaces have been filled with content, allow the user to
+        // press the button to add this entry to the list.
+        if (!nameSpace.getText().isEmpty() && !phoneSpace.getText().isEmpty() && !emailSpace.getText().isEmpty()) {
+            contacts.add(new Contact(nameSpace.getText(), phoneSpace.getText(), emailSpace.getText()));
+            nameSpace.setText("");
+            phoneSpace.setText("");
+            emailSpace.setText("");
+        }
+    }
+
+    public void removeItem() {
+        Contact contact = (Contact) list.getSelectionModel().getSelectedItem();
+        contacts.remove(contact);
+    }
+
+    public void onKeyPressed(KeyEvent event) {
+        if (event.getCode() == KeyCode.ENTER) {
+//            initialize();
+        }
+    }
+
+
+    @Override   //what are we overriding here? what are we initializing here?
+    public void initialize(URL location, ResourceBundle resources){
+        list.setItems(contacts);
+    }
+
+
+    File jsonFile = new File("jsonFormattedObjectArray.json");
+
+    JsonSerializer serializer = new JsonSerializer();
+    String jsonFormattedOjectArray = serializer.serialize(contacts);
+
```

Comment:
Good job Eric.  Nice attempt at the optional part.  You are missing a couple lines, you actually need  a filewriter object to write the file and you need to have this happen in a method that is getting called.